functionalprogramming/readme.md

```diff
@@ -0,0 +1,17 @@
+# Functional Programming in Python
+
+Functional programming is a programming paradigm that treats computation as the evaluation of mathematical functions and avoids changing-state and mutable data. In Python, we can use list comprehensions and set comprehensions as part of the functional programming approach.
+
+List comprehensions allow us to create a new list based on an existing list. Here is an example:
```

Comment:
before providing an example to explain the theory a little bit more for beginners so that it would be easy for them to catch and also provide the syntax with the explanation.